WordPress/src/main/java/org/wordpress/android/ui/uploads/UploadUtils.java

```diff
@@ -50,4 +64,151 @@
+                        }
+                    });
+            return;
+        }
+
+        boolean isScheduledPost = post != null && PostStatus.fromPost(post) == PostStatus.SCHEDULED;
+        if (isScheduledPost) {
+            // if it's a scheduled post, we only want to show a "Sync" button if it's locally saved
+            if (savedLocally) {
+                showSnackbar(activity, R.string.editor_post_saved_locally, R.string.button_sync, publishPostListener);
+            }
+            return;
+        }
+
+        boolean isPublished = post != null && PostStatus.fromPost(post) == PostStatus.PUBLISHED;
+        if (isPublished) {
+            // if it's a published post, we only want to show a "Sync" button if it's locally saved
+            if (savedLocally) {
+                showSnackbar(activity, R.string.editor_post_saved_locally, R.string.button_sync, publishPostListener);
+            } else {
+                showSnackbar(activity, activity.getString(R.string.editor_uploading_post));
+            }
+            return;
+        }
+
+        boolean isDraft = post != null && PostStatus.fromPost(post) == PostStatus.DRAFT;
+        if (isDraft) {
+            if (PostUtils.isPublishable(post)) {
+                // if the post is publishable, we offer the PUBLISH button
+                if (savedLocally) {
+                    showSnackbar(activity, R.string.editor_draft_saved_locally, R.string.button_publish, publishPostListener);
+                }
+                else {
+                    if (UploadService.hasPendingOrInProgressMediaUploadsForPost(post) ||
+                            UploadService.isPostUploadingOrQueued(post)) {
+                        showSnackbar(activity, activity.getString(R.string.editor_uploading_post));
+                    } else {
+                        showSnackbar(activity, R.string.editor_draft_saved_online, R.string.button_publish, publishPostListener);
+                    }
+                }
+            } else {
+                showSnackbar(activity, activity.getString(R.string.editor_draft_saved_locally));
+            }
+        } else {
+            if (savedLocally) {
+                showSnackbar(activity, R.string.editor_post_saved_locally, R.string.button_publish, publishPostListener);
+            }
+            else {
+                if (UploadService.hasPendingOrInProgressMediaUploadsForPost(post) ||
+                        UploadService.isPostUploadingOrQueued(post)) {
+                    showSnackbar(activity, activity.getString(R.string.editor_uploading_post));
+                } else {
+                    showSnackbar(activity, R.string.editor_post_saved_online, R.string.button_publish, publishPostListener);
+                }
+            }
+        }
+    }
+
+    public static void showSnackbar(Activity activity, int messageRes, int buttonTitleRes, View.OnClickListener onClickListener) {
```

Comment:
These two `showSnackbar()` methods can be private, I think it makes to hide them.